lib/services/elasticsearch.js

```diff
@@ -38,6 +38,60 @@ function ElasticSearch (kuzzle, options, config) {
     }
   });
 
+  this.errorMessagesMapping = [
+    {
+      // [illegal_argument_exception] object mapping [titi] can't be changed from nested to non-nested
+      regex: /^\[illegal_argument_exception\] object mapping \[(.*?)\] can't be changed from nested to non-nested$/,
+      replacement: 'Can not change mapping for field "$1" from nested to another type'
```

Comment:
Nested is a real type and it's not acting like Object type: 
https://www.elastic.co/guide/en/elasticsearch/reference/2.2/nested.html
https://www.elastic.co/guide/en/elasticsearch/reference/2.2/object.html